satpy/readers/abi_l1b.py

```diff
@@ -139,13 +147,28 @@ def _vis_calibrate(self, data):
         res.attrs['standard_name'] = 'toa_bidirectional_reflectance'
         return res
 
+    def _get_minimum_radiance(self, data):
+        """Estimate minimum radiance from Rad DataArray."""
+        attrs = data.attrs
+        scale_factor = attrs["scale_factor"]
+        add_offset = attrs["add_offset"]
+        count_zero_rad = - add_offset / scale_factor
+        count_pos = np.ceil(count_zero_rad)
+        min_rad = count_pos * scale_factor + add_offset
+        return min_rad
+
     def _ir_calibrate(self, data):
         """Calibrate IR channels to BT."""
         fk1 = float(self["planck_fk1"])
         fk2 = float(self["planck_fk2"])
         bc1 = float(self["planck_bc1"])
         bc2 = float(self["planck_bc2"])
 
+        if self.clip_negative_radiances:
+            min_rad = self._get_minimum_radiance(data)
+            clip_mask = np.logical_and(data < min_rad, ~np.isnan(data))
+            data = data.where(~clip_mask, min_rad)
```

Comment:
`data` is a `DataArray` right? So this `where` is saying "keep the existing values where "<condition>" is true, else use this other value". So if the data is already NaN then we should be fine to keep it. With this in mind, your condition could be `~(data < min_rad)` or `(data >= min_rad) && data.isnull()` since a `NaN` will always be `False` for any condition. Oh or could you just use `.clip`?

https://docs.xarray.dev/en/stable/generated/xarray.DataArray.clip.html

Reply:
`data` is a DataArray yes ... and dumb me that I didn't think to use `DataArray.clip` @djhoese. I update stuffs this afternoon